torch_geometric/nn/dense/linear.py

```diff
@@ -276,6 +281,16 @@ def forward(self, x: Tensor, type_vec: Tensor) -> Tensor:
                 out[mask] = lin(x[mask])
         return out
 
+    @torch.no_grad()
+    def initialize_parameters(self, module, input):
+        if is_uninitialized_parameter(self.weight):
+            self.in_channels = max([i.size(-1) for i in input])
```

Comment:
Can we actually test this? Why do we need the `max` here?

Reply:
sorry u are correct i was working late into the evening and forgot for a sec that this heterolinear is taking in a single tensor input since the toheterolinear i was working on is taking dict. lol just a brain fry from late nights 😝